src/uu/tail/src/args.rs

```diff
@@ -357,9 +395,27 @@ fn parse_num(src: &str) -> Result<Signum, ParseSizeError> {
     })
 }
 
-pub fn parse_args(args: impl uucore::Args) -> UResult<Settings> {
-    let matches = uu_app().try_get_matches_from(arg_iterate(args)?)?;
-    Settings::from(&matches)
+pub fn parse_args(mut args: impl uucore::Args) -> UResult<Settings> {
+    let first = args.next().unwrap();
+    let second = args.next();
+    if let Some(second) = second {
```

Comment:
Yep, much better. Yes, the MSRV doesn't enable it yet - failed some tests because of it :grin: I will adapt it. It's also easier to switch to the let else syntax, once the MSRV is high enough.

Reply:
Yeah, I love `let ... else` and would bump the MSRV if rustfmt would support it. For now, it just crashes rustfmt which is a shame.